export/export_pipeline/compute_collision_representation.py

```diff
@@ -0,0 +1,427 @@
+import subprocess
+import trimesh
+import utils
+import yaml
+
+
+osp = os.path
+
+
+def decimation_lookup(poly_count):
+    # Get the adaptive decimation ratio
+    if poly_count < 10000:
+        return 1.0
+    elif poly_count < 50000:
+        return 0.8
+    elif poly_count < 100000:
+        return 0.6
+    elif poly_count < 500000:
+        return 0.5
+    elif poly_count < 1000000:
+        return 0.2
+    elif poly_count < 5000000:
+        return 0.1
+    elif poly_count < 10000000:
+        return 0.01
+    elif poly_count < 20000000:
+        return 0.009
+    elif poly_count < 30000000:
+        return 0.008
+    elif poly_count < 40000000:
+        return 0.009
+    elif poly_count < 50000000:
+        return 0.005
+    else:
+        return 0.001
+
+
+def get_relative_pose(obj):
+    """
+    gets pose relative to parent
+    """
+    obj.select_set(True)
+    xyz = copy.deepcopy(obj.location)
+    xyz = [xyz.x, xyz.y, xyz.z]
+    quat = copy.deepcopy(obj.rotation_quaternion)
+    quat = [quat.w, quat.x, quat.y, quat.z]
+    obj.select_set(False)
+    return xyz, quat
+
+
+def get_relative_pose_and_center(obj, scale_factor=np.ones(3)):
+    """
+    gets pose relative to parent, and centers obj w.r.t. parent
+    """
+    xyz, quat = get_relative_pose(obj)
+    xyz = np.asarray(xyz) * scale_factor
+    obj.select_set(True)
+    obj.location = (0.0, 0.0, 0.0)
+    obj.rotation_quaternion = (1.0, 0.0, 0.0, 0.0)
+    obj.select_set(False)
+    return xyz.tolist(), quat
+
+
+def process_mesh(args: tuple) -> bool:
+    """
+    Decompose a mesh into convex regions according to hyperparameters.
+    Save the convex regions as well as a convex hull of the mesh in a directory structure.
+    args: a tuple containing (path to mesh, hyperparameters, rerun flag).
+    Returns a bool indicating success of the operation.
+    """
+    mesh_path, params, rerun = args
+    mesh_dir, mesh_filename = osp.split(mesh_path)
+    vhacd_args = params['collision_representation']['vhacd']
+    coacd_args = params['collision_representation']['coacd']
+    acd_args   = params['collision_representation']['acd']
+
+    # Copy the obj file to the temporary directory.
+    decomposed = False
+    decompose_dir = osp.join(mesh_dir, params['common']['CONVEX_DECOMPOSITION_DIR'])
+    if rerun and osp.isdir(decompose_dir):
+        decomposed = True
+        os.remove(mesh_path)
+    else:
+        os.makedirs(decompose_dir, exist_ok=True)
+        mesh_path = shutil.move(mesh_path, decompose_dir)
+        obj_name = os.path.splitext(mesh_filename)[0]
+        
+        # load mesh and check that it is valid
+        mesh: trimesh.Trimesh = trimesh.load(mesh_path, force='mesh')
+        if mesh.is_empty:
+            volume = 0.0
+        else:
+            try:
+                volume = mesh.bounding_box_oriented.volume
+            except ValueError:
+                volume = 0.0
+        if volume < 1e-6:
+            acd_args['decompose_method'] = 'skip'
```

Comment:
What is the difference between `skip` and `none`? Our convention is to treat config values as read-only unless there is a good reason not to.